packages/@aws-cdk/aws-servicecatalog/lib/product-stack.ts

```diff
@@ -50,6 +69,44 @@ export class ProductStack extends cdk.Stack {
     return cdk.Lazy.uncachedString({ produce: () => this._templateUrl });
   }
 
+  /**
+   * Fetch the asset bucket.
+   *
+   * @internal
+   */
+  public _getAssetBucket(): IBucket | undefined {
+    return this.assetBucket;
+  }
+
+  /**
+   * Asset are prepared for bulk deployment to S3.
+   * @internal
+   */
+  public _addAsset(asset: cdk.FileAssetSource): void {
+    const assetPath = './cdk.out/' + asset.fileName;
+    this.assets.push(Source.asset(assetPath));
+  }
+
+  /**
+   * Deploy all assets to S3.
+   * @internal
+   */
+  private _deployAssets() {
```

Comment:
See my other comment about doing this within the synthesizer

Reply:
I moved them both over as suggested.